engine/deck.go

```diff
@@ -1,17 +1,35 @@
 package engine
 
-import "errors"
+import (
+	"errors"
+	"math/rand"
+	"time"
+)
 
-// Deal takes a pointer to deck (slice) of Cards and returns a slice of numDealt 
-// Cards from it. If the slice of cards is not big enough, it returns an empty 
+// Deal takes a pointer to deck (slice) of Cards and returns a slice of numDealt
+// Cards from it. If the slice of cards is not big enough, it returns an empty
 // slice and an error. Deck is reduced in size by numDealt.
-
-func Deal (numDealt uint, deck *[]Card) (hand []Card, err error) {
+func Deal(numDealt uint, deck *[]Card) (hand []Card, err error) {
 	d := *deck
 	if len(d) < int(numDealt) {
 		return hand, errors.New("Deal: not enough cards in deck")
 	}
 	hand = d[:numDealt]
 	*deck = d[numDealt:]
 	return
-}
\ No newline at end of file
+}
+
+func shuffle(cards []Card) {
+	src := rand.NewSource(time.Now().UnixNano())
+	gen := rand.New(src)
+	for i := len(cards) - 1; i >= 0; i-- {
+		rIdx := gen.Intn(327) % (i + 1)
+		swap(cards[:], rIdx, i)
+	}
+}
+
+func swap(deck []Card, i int, j int) {
```

Comment:
If we don't export this maybe a private helper function comment over it would be nice? I like that you did that in the other modules